diff --git a/Makefile b/Makefile
index 70cc7b1..66cb6f2 100644
--- a/Makefile
+++ b/Makefile
@@ -34,7 +34,7 @@ SONAME_MAJOR := $(word 1,$(subst ., ,$(VERSION)))
 SONAME_MINOR := $(word 2,$(subst ., ,$(VERSION)))

 # OS-specific bits
-ifeq ($(shell uname),Darwin)
+ifeq ($(findstring darwin, $(shell $(CC) -dumpmachine)), darwin)
 	SOEXT = dylib
 	SOEXTVER_MAJOR = $(SONAME_MAJOR).$(SOEXT)
 	SOEXTVER = $(SONAME_MAJOR).$(SONAME_MINOR).$(SOEXT)
