diff --git a/Makefile b/Makefile
index e021e87..fb9d2ef 100644
--- a/Makefile
+++ b/Makefile
@@ -30,7 +30,7 @@ SONAME_MINOR := $(word 2,$(subst ., ,$(VERSION)))
 # OS-specific bits
 ifeq ($(OS),Windows_NT)
 	$(error "Windows is not supported")
-else ifeq ($(shell uname),Darwin)
+else ifeq ($(findstring darwin, $(shell $(CC) -dumpmachine)), darwin)
 	SOEXT = dylib
 	SOEXTVER_MAJOR = $(SONAME_MAJOR).dylib
 	SOEXTVER = $(SONAME_MAJOR).$(SONAME_MINOR).dylib
